src/Fantomas.Core/CodePrinter.fs

```diff
@@ -308,6 +308,31 @@ let isSynExprLambdaOrIfThenElse (e: Expr) =
     | Expr.IfThenElse _ -> true
     | _ -> false
 
+let (|RecordOrAnonRecord|_|) (expr: Expr) =
```

Comment:
I'm not really a fan of this active pattern. It somewhat ruins the magic that you access an exact node type from `SyntaxOak` in `CodePrinter`.
The anonymous records used are also not consistent with anything else in this codebase.
If you were going to make this split, it should be in `SyntaxOak` where we have a shared `Node` for the two when possible.

Reply:
That makes a lot of sense. My original plan was essentially doing what my active pattern did but with a single shared Node, but I had issues because of something about trivia which led me to try the active pattern instead. But consolidating via Node inheritance makes a ton of sense. My goal was mainly to reduce duplication and make it easier for one change to fix a problem in both, but I also don't want to sacrifice readability or make anything harder to follow. Also didn't love having to have additional cases that failed because the exhaustive check failed. I'll try and refactor to your approach here instead.